unittests/Threading/Thread.cpp

```diff
@@ -0,0 +1,27 @@
+//===--- Thread.cpp - Tests swift::Thread functionality -------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Threading/Thread.h"
+#include "gtest/gtest.h"
+
+using namespace swift;
+
+// Check that Thread::stackBounds() returns a reasonable value or none at all.
+TEST(StackTest, stack_bounds) {
+  auto bounds = Thread::stackBounds();
+
+  if (bounds.has_value()) {
+    ASSERT_TRUE(bounds->low != nullptr);
+    ASSERT_TRUE(bounds->high != nullptr);
+    ASSERT_TRUE(bounds->low < bounds->high);
```

Comment:
I'd recommend casting the pointers to `uintptr_t` before comparing them here, since if the function is broken and returns bad pointers, or one is `nullptr`, the result will be undefined.

Reply:
Good point.  Done, thanks.